main/tasks/statistics_task.c

```diff
@@ -23,7 +23,7 @@ static uint16_t statisticsDataSize;
 static pthread_mutex_t statisticsDataLock = PTHREAD_MUTEX_INITIALIZER;
 
 static const uint16_t maxDataCount = 720;
-static uint16_t statsFrequency;
+static uint32_t statsFrequency;
```

Comment:
This doesn't need to be a static field, as it's only used in the while loop. Might as well just define it there.

Maybe for clarity, get the value from NVS into a `uint16_t`, but then do the `* 1000` into a `uint32_t`.